File: Data Structures & Algorithms/binary-tree-right-side-view/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {

    ArrayList<Integer> res = new ArrayList<>();
    HashMap<Integer, Integer> level = new HashMap<>();
    int height = 0;

    public List<Integer> rightSideView(TreeNode root) {
        dfs(root, 0);
        for(int i = 0; i<height; i++){
            res.add(level.get(i));
        }
        return res;
    }

    private void dfs(TreeNode node, int depth){

        if(node == null){
            return;
        }

        if(!level.containsKey(depth)){
            height++;
        }

        level.put(depth, node.val);

        dfs(node.left, depth+1);
        dfs(node.right, depth+1);

    }
}

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:

    res = 0
    def goodNodes(self, root: TreeNode) -> int:
        self.dfs(root, -101)
        return self.res

    def dfs(self, node: TreeNode, maxVal: int):
        
        if(node == None):
            return 

        if(node.val >= maxVal):
            maxVal = node.val
            self.res += 1

        self.dfs(node.left, maxVal)
        self.dfs(node.right, maxVal)




        
        